external-import/socradar/src/lib/radar.py

```diff
@@ -0,0 +1,354 @@
+            maintainer_identity = self._get_or_create_identity(maintainer)
+            if not maintainer_identity:
+                self.helper.log_error(
+                    f"Could not create identity for maintainer: {maintainer}"
+                )
+                return
+
+            pattern = self._create_stix_pattern(value, feed_type)
+            if not pattern:
+                self.helper.log_error(
+                    f"Could not create pattern for: {value} ({feed_type})"
+                )
+                return
+
+            # Create kill chain phase
+            kill_chain_phase = KillChainPhase(
+                kill_chain_name="lockheed-martin-cyber-kill-chain",
+                phase_name="reconnaissance",
+            )
+
+            indicator = Indicator(
+                id=f"indicator--{str(uuid.uuid4())}",
+                name=f"{feed_type.upper()}: {value}",
+                description=f"Type: {feed_type}\nValue: {value}\nSource: {maintainer}",
+                pattern_type="stix",
+                pattern=pattern,
+                valid_from=first_seen,
+                valid_until=last_seen,
+                labels=[feed_type, "malicious-activity"],
+                confidence=75,
+                indicator_types=self._get_indicator_type(feed_type),
+                kill_chain_phases=[kill_chain_phase],
+                created=first_seen,
+                modified=first_seen,
+                created_by_ref=maintainer_identity.id,  # Use maintainer's identity
+                object_marking_refs=[TLP_WHITE],
+            )
+
+            # Create relationship between indicator and maintainer identity
+            relationship = Relationship(
+                id=f"relationship--{str(uuid.uuid4())}",
+                relationship_type="created-by",
+                source_ref=indicator.id,
+                target_ref=maintainer_identity.id,
+                description=f"This indicator was created by {maintainer}",
+                created=first_seen,
+                modified=first_seen,
+                confidence=75,
+                object_marking_refs=[TLP_WHITE],
+            )
+
+            # Create bundle with all objects
+            bundle = Bundle(objects=[maintainer_identity, indicator, relationship])
+
+            # Send to OpenCTI
+            self.helper.send_stix2_bundle(bundle.serialize(), work_id=work_id)
+
+            self.helper.log_info(
+                f"Created {feed_type} indicator for: {value} from {maintainer}"
+            )
+
+        except Exception as e:
+            self.helper.log_error(f"Error processing item {str(item)}: {str(e)}")
+
+    def _create_observable(self, value, feed_type):
+        """Create appropriate observable based on value type with proper STIX ID"""
+        try:
+            if feed_type == "url" or self._matches_pattern(value, "url"):
+                return URL(
+                    id=f"url--{str(uuid.uuid4())}",
```

Comment:
stix2.Observable IDs are deterministic if not explicitly modified.
This allows the platform to de-duplicate this entities.